### DISCUSSION TITLE
How to configure CORS in gunicorn

### DISCUSSION THREAD

**Discussion by jtvlrm** [benoitc/gunicorn#3001] [Q&A] [answered]
I have a setup where I'm using a web server on the domain [www.example.com](http://www.example.com/), and running Gunicorn (without using any framework like Flask...) on [www.example.com:8888](http://www.example.com:8888/). 

I have a page, www.example.com/index.html, where I make a fetch request to www.example.com:8888. 
However, I'm getting a CORS error: "Access to fetch at 'https://www.example.com:8888/' from origin 'https://example.com/' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource." 

If I do it from localhost, there is no problem. (Using 127.0.0.1/iindex.html y 127.0.0.1:8888
Is it possible to control the CORS configuration in Gunicorn? I would like to set a configuration on example.com:8888 (Gunicorn) of this type: Access-Control-Allow-Origin: www.example.com.

Thanks in Advance.

**Comment by jtvlrm** [comment, marked as answer]
Well, I have the answer. 
The issue is that in browsers, when there are cross-domain requests, they usually send a preflight request to check the CORS configuration of the server. What needs to be done in the Python application is to add the following code at the beginning of the file. This will handle the response appropriately.

    if environ['REQUEST_METHOD'] == 'OPTIONS':
        # Respuesta para las solicitudes preflight (OPTIONS)
        response_headers = [ ('Access-Control-Allow-Origin', '*'), ('Access-Control-Allow-Methods', 'POST, OPTIONS'), ('Access-Control-Allow-Headers', 'Content-Type') ]
        start_response('200 OK', response_headers)
        print("a responder")
        return []

**Reply by benoitc** [reply]
yes that should be done at atpplication level. I would suggest to use a wsgi middleware for it
